### PR TITLE
Remove use of `String.Split` in Environment taghelper

### LINKED ISSUE #3617: Remove use of `String.Split` in Environment taghelper
image

Use `StringSegment` or similar

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3617

### CODE REVIEW

**Comment by pranavkm** [comment]
## Before
envtaghelper - before
envtaghelper - before1

## After
envtaghelper - after

This accounts for 670k (~1.2% of allocations) over 3000 requests. This was tested using TagHelperSample.Web

**Comment by pranavkm** [comment]
cc @dougbu \ @NTaylorMullen

File: src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs [hunk 1]
```diff
@@ -74,36 +68,28 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 return;
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs]: Not really related to this PR but this `return` is for a weird case. Very odd that `<environment Names="dev">content</environment>` writes out `content` when `EnvironmentName` is `string.Empty` but not when the property has any other value (but `"dev"`).

**Comment by dougbu** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**